### PR TITLE
IGNITE-18205 Sql. Allow hash output distribution for aggregations

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/IgniteColocatedSortAggregate.java [hunk 1]
```diff
@@ -39,15 +39,15 @@
  * IgniteSingleSortAggregate.
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/IgniteColocatedSortAggregate.java:39]: rename

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/IgniteColocatedAggregateBase.java [hunk 2]
```diff
@@ -36,9 +38,9 @@
  * IgniteSingleAggregateBase.
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/agg/IgniteColocatedAggregateBase.java:38]: rename

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/TableDescriptorImpl.java [hunk 3]
```diff
@@ -127,7 +135,7 @@ public RelDataType deleteRowType(IgniteTypeFactory factory) {
     /** {@inheritDoc} */
     @Override
     public IgniteDistribution distribution() {
-        return IgniteDistributions.random();
+        return IgniteDistributions.affinity(colocationColumns, 0, dummyAffinityIdentity);
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/TableDescriptorImpl.java:138]: why we use 0 here?

Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/TableDescriptorImpl.java]: I've added comment explaining such choice. Also check please comment in beginning of the class

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**